notifications-sandbox/index.js

```
@@ -0,0 +1,34 @@
+const { ENVIRONMENT } = process.env
+const https = require('https')
+
+const log = require('SyntheticsLogger')
+const smokeTestHelpers = require('../helpers/smokeTestHelpers')
+
+exports.handler = async () => {
+  const secret = await smokeTestHelpers.getSecret(`${ENVIRONMENT}/smoke_test`)
+  const apiToken = secret.NOTIFICATIONS_SANDBOX_AUTH_TOKEN
+  const notificationsHostName = secret.NOTIFICATIONS_SANDBOX_HOST
+  const path = '/v1/api/notifications/sandbox'
+
+  log.info(`Sending POST request to ${notificationsHostName}${path}`)
+  const options = {
+    host: notificationsHostName,
+    port: 443,
+    headers: {
+      Authorization: apiToken
+    },
+    path,
+    method: 'POST'
+  }
+
+  return new Promise((resolve, reject) => {
+    https.request(options, res => {
+      if (res.statusCode === 200) {
+        log.info('Notifications responded with 200 OK')
+        resolve()
+      } else {
+        reject(new Error(`Notifications endpoint responded with ${res.statusCode} status`))
+      }
+    }).end()
+  })
```

Do we want to wrap the request in `synthetics.executeStep` so it shows up neatly in the Canary dashboard?

---

I did consider it but then thought it best to keep it simple. It really only does one thing, and doesn't have multiple steps (also there's no output like screen grabs to organise by steps etc).

---

Just thinking about how we can debug failures - when the canaries were failing the other week, we only got a basic '429 status' error rather than a traceback, because the canary hadn't reached the first `executeStep` yet. Will the `Notifications endpoint responded with...` error message ever actually show up in the console?

---

yeah it should do but that is something I want to confirm when I run it as a Canary and we take another look. Just added the synethics step and it adds another layer of async functions :(. As it stands the `reject(new Error...` will be caught in the try catch block that runs the `handler` function and it should report the error correctly.

I'll test all of those scenarios and if its not clear when it fails I can adjust it if that sounds ok?